src/H5F.c

```diff
@@ -196,15 +196,31 @@ H5Fget_access_plist(hid_t file_id)
  *-------------------------------------------------------------------------
  */
 static int
-H5F__get_all_count_cb(void H5_ATTR_UNUSED *obj_ptr, hid_t H5_ATTR_UNUSED obj_id, void *key)
+H5F__get_all_count_cb(void H5_ATTR_UNUSED *obj_ptr, hid_t obj_id, void *key)
 {
-    H5F_trav_obj_cnt_t *udata     = (H5F_trav_obj_cnt_t *)key;
-    int                 ret_value = H5_ITER_CONT; /* Return value */
+    H5F_trav_obj_cnt_t *udata        = (H5F_trav_obj_cnt_t *)key;
+    H5I_type_t          obj_type     = H5I_UNINIT;
+    htri_t              is_committed = FAIL;
+    int                 ret_value    = H5_ITER_CONT; /* Return value */
 
-    FUNC_ENTER_PACKAGE_NOERR
+    FUNC_ENTER_PACKAGE
+
+    if ((obj_type = H5Iget_type(obj_id)) < 0)
```

Comment:
I think this is a bug and should be corrected.  However, the library internal routines should never (with *very* few exceptions) call API routines.  I'll take a look at which internal routines to use later tonight.  (If someone else doesn't get to it before me :-) ).

Reply:
Replaced `H5Iget_type` -> `H5I_get_type`,  `H5Tcommitted` -> `H5I_object_verify` + `H5T_is_named`